Test/Tests/RealtimeClientConnectionTests.swift

```diff
@@ -3736,6 +4193,163 @@ class RealtimeClientConnectionTests: XCTestCase {
                 XCTAssertEqual(newTransport.protocolMessagesSent.filter { $0.action == .message }.count, 1)
                 done()
             }
+            client.connection.once(.disconnected) { _ in
+                expect(client.internal.pendingMessages).to(haveCount(1))
+                print("Disconnected");
+            }
+            // reconnected
+            client.connection.once(.connected) { _ in
+                expect(client.internal.pendingMessages).to(haveCount(1))
+            }
+            client.internal.onDisconnected()
+        }
+        // The connection should also process any messages queued per RTL6c2
+        expect(client.internal.queuedMessages).toEventually(haveCount(0), timeout: testTimeout)
```

Comment:
I've inspected RTN19a1 tests in more depth and didn't find any of them very useful. So I've removed them and instead added RTN19a1 checks to the RTN19a test. What's described in RTN19a is more suitable for what are `pendingMessages` for, but I've also added checks for `queuedMessages`.

Reply:
I've looked at the changes that you've made to `test__066...` in https://github.com/ably/ably-cocoa/commit/fa8e375f501ac2c1e511b0776707f8c2634c877c and I'm struggling to understand the motivation for the various assertions that you've added around the counts of `queuedMessages` and `pendingMessages` (that is, why, at one moment, you expect `pendingMessages` to be non-empty, and then why at a different moment you now expect `queuedMessages` to be non-empty). Is there anything you can do to help me understand this? Some comments?

Reply:
>What is the message that's being queued here

It was added in e7495cfdbfabac5a941d0a059902fa644ddefddc and I just left it there as-is... Probably was copied into from somewhere else. It shouldn't be there since RTN19a is about pending messages (not queued ones).

Reply:
[Reverted](https://github.com/ably/ably-cocoa/pull/1714/commits/99c4ec01c3299ac13d0c25dc6d490d7a1b8ed4ef) https://github.com/ably/ably-cocoa/commit/fa8e375f501ac2c1e511b0776707f8c2634c877c

now this https://github.com/ably/ably-cocoa/commit/fa8e375f501ac2c1e511b0776707f8c2634c877c